lib/axon/loop.ex

```diff
@@ -556,13 +556,16 @@ defmodule Axon.Loop do
     {type, _} = val.type
 
     case type do
-      t when t in [:s, :u] -> "#{name}: #{:io_lib.format('~.B', [Nx.to_number(val)])}"
-      :f -> "#{name}: #{:io_lib.format('~.7f', [Nx.to_number(val)])}"
-      :bf -> "#{name}: #{:io_lib.format('~.3f', [Nx.to_number(val)])}"
-      _ -> "Unsupported type of metric"
+      t when t in [:s, :u] -> "#{name}: #{Nx.to_number(val)}"
+      :f -> "#{name}: #{float_format('~.7f', Nx.to_number(val))}"
+      :bf -> "#{name}: #{float_format('~.3f', Nx.to_number(val))}"
+      _ -> "#{name}: unsupported type of metric #{inspect(type)}"
     end
   end
 
+  defp float_format(format, val) when is_atom(val), do: Atom.to_string(val)
```

Comment:
In Nx and Complex we inspect :nan, :infinity and :neg_infinity respectively as NaN, Inf and -Inf

Shouldn't we do the same here?